app/main.py

```diff
@@ -1,25 +1,98 @@
+from __future__ import annotations
 from abc import ABC
 
 
 class IntegerRange:
-    pass
+
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(
+            self,
+            owner: object,
+            name: str
+    ) -> None:
+        self.protect_name = "_" + name
+
+    def __get__(
+            self,
+            instance: object,
+            owner: object
+    ) -> int:
+        return getattr(instance, self.protect_name)
+
+    def __set__(
+            self,
+            instance: object,
+            value: int
+    ) -> None:
+        if isinstance(self.min_amount, int) \
+                and isinstance(self.max_amount, int):
+            if self.min_amount <= value <= self.max_amount:
+                setattr(instance, self.protect_name, value)
+            else:
+                raise ValueError("ValueError")
 
 
 class Visitor:
-   pass
+
+    def __init__(self, name: str, age: int, height: int, weight: int) -> None:
+        self.name = name
+        self.age = age
+        self.height = height
+        self.weight = weight
 
 
 class SlideLimitationValidator(ABC):
-    pass
+
+    def __init__(self, age: int, height: int, weight: int) -> None:
+        self.age = age
+        self.height = height
+        self.weight = weight
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+
+    age = IntegerRange(4, 14)
+    height = IntegerRange(80, 120)
+    weight = IntegerRange(20, 50)
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+
+    age = IntegerRange(14, 60)
+    height = IntegerRange(120, 220)
+    weight = IntegerRange(50, 120)
 
 
 class Slide:
-    pass
+
+    def __init__(
+            self,
+            name: str,
+            limitation_class: SlideLimitationValidator
```

Comment:
```suggestion
            limitation_class: type[SlideLimitationValidator]
```
It accept type not instance